### ISSUE TITLE
`np.sum` over an array axis giving a wrong result

### ISSUE THREAD

**Issue by rsarm** [nv-legate/legate.numpy#47]
Hi. I'm trying legate.numpy on Piz Daint. The following should give a symmetric matrix, however with legate.numpy, it gives a matrix of zeros:

```python
import legate.numpy as np


x = np.random.random((60, 30))

def euclidean_broadcast(x, y):
    """r_ij = (x_ij - y_ij)^2"""

    diff = x[:, np.newaxis, :] - y[np.newaxis, :, :]
    return (diff * diff).sum(axis=2)


edm = euclidean_broadcast(x, x)
print(edm[:3, :3])
```

I'm running with
```bash
legate --launcher srun --nodes 1 --gpus 1 --fbmem 14000  edm.py -lg:numpy:test --eager-alloc-percentage 5
```
`edm.py` is the script with code above.

Everything looks fine up to the `(diff * diff)`. It's the `sum(axis=2)` what gives the zeros. It happens when using `--gpus`. The cpu version works fine.

I'm using this commits:
```
legate.numpy 496c64d (2021-05-12)
legate.core  9e327b7 (2021-05-12)
```
Please, let me know if you need more information. Thanks in advance!

**Comment by manopapad** [comment]
This should be fixed with 05bf3ecd, could you please try again?

**Comment by rsarm** [comment]
Yes, that solves the issue! Thank you very much!
